Gems/ROS2/Code/Source/Manipulator/JointPublisherComponent.cpp

```diff
@@ -0,0 +1,126 @@
+#include <ROS2/Manipulator/JointPublisherComponent.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <ROS2/ROS2Bus.h>
+#include <PhysX/Joint/PhysXJointRequestsBus.h>
+#include <Source/HingeJointComponent.h>
+#include <AzCore/Component/ComponentApplicationBus.h>
+#include <ROS2/Frame/ROS2FrameComponent.h>
+#include <RobotImporter/URDFMetadataComponent.h>
+#include <ROS2/Utilities/ROS2Names.h>
+
+namespace ROS2
+{
+    void JointPublisherComponent::Activate()
+    {
+        AZ::TickBus::Handler::BusConnect();
+        auto ros2Node = ROS2::ROS2Interface::Get()->GetNode();
+        auto ros2Frame = GetEntity()->FindComponent<ROS2FrameComponent>();
+        AZStd::string namespacedTopic = ROS2Names::GetNamespacedName(ros2Frame->GetNamespace(), "joint_states");
+        m_jointstatePublisher = ros2Node->create_publisher<sensor_msgs::msg::JointState>(namespacedTopic.data(), rclcpp::QoS(1));        // TODO: add QoS instead of "1"
```

Comment:
In general we don't want to tag the code with // TODO comments. Instead, create a GHI to add QoS here in the future instead